checker/bin-devel/test-cftests-junit-jdk21.sh

```diff
@@ -12,4 +12,4 @@ export ORG_GRADLE_PROJECT_useJdkCompiler=21
 source "$SCRIPTDIR"/clone-related.sh
 
 
-./gradlew test -x javadoc -x allJavadoc --console=plain --warning-mode=all
+./gradlew test -x javadoc -x allJavadoc --console=plain --warning-mode=all --max-workers=1
```

Comment:
In #858 I'm trying to disable the configuration cache instead. That is what we did manually to fix the flakiness, so hopefully that also works here.
There is some connectedness here, too: https://docs.gradle.org/current/userguide/performance.html#additional_configuration_cache_benefits says that the configuration cache `Executes all tasks in parallel`. So maybe disabling the configuration cache does more than just setting the max-workers to 1.

[This doc](https://docs.gradle.org/current/userguide/command_line_interface.html#sec:command_line_performance) says that by default the configuration cache is off, but probably our CI setup enables it somehow.
That doc also says that `--parallel` is off by default, but it's not clear to me what the relation between `max-workers` and `parallel` is... Are the workers still used in parallel even if `parallel` is off? Strange...

But maybe the combination here is better. Anyways, there is more documentation here https://docs.gradle.org/current/userguide/configuration_cache.html and https://docs.gradle.org/current/userguide/performance.html#enable_configuration_cache

I also note that in typetools PRs 6422 and 6564 the `maxParallelForks` option is set for tests instead. There is some Azure-specific logic. It's again unclear what the relation between `parallel forks` and `max workers` is.

Reply:
From #858 it seems to be the build cache, not the configuration cache, that is causing the problems. So is it better to only have 1 worker or have the build cache? Can you compare timing between the two fixes?

Reply:
> From #858 it seems to be the build cache, not the configuration cache, that is causing the problems. So is it better to only have 1 worker or have the build cache? Can you compare timing between the two fixes?

sure, maybe let's wait the ci finishes and do rerun a few times. We can see the how long does the junit, non-junit and junit java 21 task takes.

Reply:
Both this PR and #858 take about 11h 42m... Let's just keep the solution here.
Can you update the .sh files and add a comment before why we're adding the flag, maybe that disabling the build cache would be an alternative.
If this closes issue #849, let's add a changelog entry. Alternatively, we can keep #849 open and see whether we fix the actual build issue within dataflow tests at some point.

Reply:
>Can you update the .sh files and add a comment before why we're adding the flag, maybe that disabling the build cache would be an alternative.

Sure.
>If this closes issue https://github.com/eisop/checker-framework/issues/849, let's add a changelog entry. Alternatively, we can keep https://github.com/eisop/checker-framework/issues/849 open and see whether we fix the actual build issue within dataflow tests at some point.

I prefer to keep #849. I think the reason might be some task should be execute before `dataflow:busyexpression`.